agent/engine/docker_task_engine.go

```diff
@@ -449,11 +450,45 @@ func (engine *DockerTaskEngine) checkTaskState(task *apitask.Task) {
 					Status:                  status,
 					DockerContainerMetadata: metadata,
 				},
+				source: fromInspect,
+				containerPrevRestartAttempts: restartAttempts,
 			})
 		}
 	}
 }
 
+// checkTaskState inspects the state of input container and writes
+// its state to the managed task's container channel.
+func (engine *DockerTaskEngine) checkContainerState(container *apicontainer.Container, task *apitask.Task) {
+	taskContainers, ok := engine.state.ContainerMapByArn(task.Arn)
+	if !ok {
+		seelog.Warnf("Task engine [%s]: could not check container %s state; no task in state", task.Arn, container.Name)
+		return
+	}
+
+	dockerContainer, ok := taskContainers[container.Name]
+	if !ok {
+		return
+	}
+	restartAttempts := container.GetRestartAttempts()
+	status, metadata := engine.client.DescribeContainer(engine.ctx, dockerContainer.DockerID)
+	engine.tasksLock.RLock()
+	managedTask, ok := engine.managedTasks[task.Arn]
+	engine.tasksLock.RUnlock()
+
+	if ok {
```

Comment:
if !ok, nothing should happen?

Can we throw an error if !ok?

Reply:
If !ok, the task of this container is not tracked by Agent. If this happens, I think Agent couldn't do much about this and can just ignore it.